app/controllers/api/v1/search_controller.rb

```diff
@@ -291,15 +291,18 @@ def index
           end
         end
 
-        # perform TDR search, if enabled & user performed faceted search
-        if @facets.any? && User.feature_flag_for_instance(current_api_user, 'cross_dataset_search_backend')
-          @trd_results = self.class.get_tdr_results(@facets)
+        # perform TDR search, if enabled, and there are facets/terms provided by user
+        if User.feature_flag_for_instance(current_api_user, 'cross_dataset_search_backend') && (@facets.present? || @term_list.present?)
+          @tdr_results = self.class.get_tdr_results(selected_facets: @facets, terms: @term_list)
           # just log for now
-          logger.info "Found #{@trd_results.keys.size} results in Terra Data Repo"
-          logger.info pp @trd_results
+          logger.info "Found #{@tdr_results.keys.size} results in Terra Data Repo"
+          logger.info pp @tdr_results if @tdr_results.any?
+          @tdr_results.each do |short_name, tdr_result|
+            @studies << tdr_result
+          end
         end
 
-        @matching_accessions = @studies.map(&:accession)
+        @matching_accessions = @studies.map {|study| study.is_a?(Study) ? study.accession : study[:accession]}
```

Comment:
FYI: this could be replaced with just `@studies.map{ |study| study[:accession] }`  but it might be clearer the way you have it that 'hey, there are two different kinds of things in this list'